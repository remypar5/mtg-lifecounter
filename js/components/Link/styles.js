import { StyleSheet } from 'react-native';
import { COLOR_MARKED } from '../../utils/constants';

const styles = StyleSheet.create({
    link: {
        color: COLOR_MARKED,
    },
});

export default styles;
