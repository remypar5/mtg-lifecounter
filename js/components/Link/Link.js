import React from 'react';
import PropTypes from 'prop-types';
import { Linking, Text as NativeText } from 'react-native';

// import Text from '../Text';
import { StylePropType } from '../../types';
import { flattenStyles } from '../../utils';

import styles from './styles';

const { link } = styles;

const openURL = (url) => Linking.canOpenURL(url)
    .then((supported) => {
        if (supported) {
            return Linking.openURL(url);
        }
        return Promise.reject(url);
    });

const Link = ({ href, children, style }) => (
    <NativeText
        onPress={() => openURL(href)}
        style={flattenStyles(link, style)}
    >
        {children}
    </NativeText>
);

Link.propTypes = {
    href: PropTypes.string.isRequired,
    children: PropTypes.string.isRequired,
    style: StylePropType,
};

Link.defaultProps = {
    style: null,
};

export default Link;
// export {
//     openURL,
// };
