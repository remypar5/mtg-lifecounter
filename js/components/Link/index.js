import Link from './Link';
import styles from './styles';

export default Link;
export {
    styles,
};
