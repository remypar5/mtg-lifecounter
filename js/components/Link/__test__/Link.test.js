import React from 'react';
import { Linking } from 'react-native';
import { shallow } from 'enzyme';

import Link from '../Link';


jest.mock('Linking', () => ({
    canOpenURL: jest.fn()
        .mockReturnValue({
            then: () => Promise.resolve(true),
        }),
    openURL: jest.fn()
        .mockReturnValue({
            then: () => Promise.resolve(),
        }),
}));

describe('<Link />', () => {
    let component;
    const canOpenURLSpy = jest.spyOn(Linking, 'canOpenURL')
        .mockImplementation(() => Promise.resolve(true));
    const openURLSpy = jest.spyOn(Linking, 'openURL')
        .mockImplementation(() => Promise.resolve());

    beforeAll(() => {
        canOpenURLSpy.mockClear();
        openURLSpy.mockClear();

        component = shallow((
            <Link href="https://github.com/remypar5/mtg-lifecounter">
                Fork me on GitHub
            </Link>
        ));
    });

    it('renders without crashing', () => {
        expect(component).toMatchSnapshot();
    });

    it('tries to open the given `href`', () => {
        expect.assertions(1);
        const link = component.find('Text').at(0);
        link.simulate('press');

        // console.log(canOpenURLSpy);

        return expect(canOpenURLSpy()).resolves.toBe(true);
    });
});
