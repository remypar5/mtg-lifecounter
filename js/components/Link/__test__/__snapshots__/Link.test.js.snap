// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Link
    href="https://github.com/remypar5/mtg-lifecounter"
    style={null}
  >
    Fork me on GitHub
  </Link>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "accessible": true,
      "allowFontScaling": true,
      "children": "Fork me on GitHub",
      "ellipsizeMode": "tail",
      "onPress": [Function],
      "style": Object {
        "color": "#710000",
      },
    },
    "ref": null,
    "rendered": "Fork me on GitHub",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Fork me on GitHub",
        "ellipsizeMode": "tail",
        "onPress": [Function],
        "style": Object {
          "color": "#710000",
        },
      },
      "ref": null,
      "rendered": "Fork me on GitHub",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
